File: API/Models/MS_QR_Order.cs

using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Models
{
    public partial class MS_QR_Order
    {
        [Key]
        [StringLength(1)]
        public string manuf { get; set; }
        [Required]
        [StringLength(10)]
        public string brandname { get; set; }
        [Required]
        [StringLength(40)]
        public string cusid { get; set; }
        [Required]
        [StringLength(10)]
        public string cusna { get; set; }
        [Required]
        [StringLength(1)]
        public string kind { get; set; }
        [Key]
        [StringLength(10)]
        public string purno { get; set; }
        [Key]
        [StringLength(15)]
        public string manno { get; set; }
        [Column(TypeName = "date")]
        public DateTime? eta { get; set; }
        [Required]
        [StringLength(16)]
        public string bitnbr { get; set; }
        [Required]
        [StringLength(10)]
        public string ritnbr { get; set; }
        [Required]
        [StringLength(15)]
        public string rmodel { get; set; }
        [StringLength(3)]
        public string tolcls { get; set; }
        [Required]
        [StringLength(6)]
        public string article { get; set; }
        [StringLength(15)]
        public string style { get; set; }
        [Key]
        [StringLength(4)]
        public string size { get; set; }
        [Required]
        [StringLength(4)]
        public string tsize { get; set; }
        [Column(TypeName = "decimal(7, 1)")]
        public decimal qty { get; set; }
        [Column(TypeName = "decimal(7, 1)")]
        public decimal pqty { get; set; }
        [Column(TypeName = "decimal(7, 1)")]
        public decimal cqty { get; set; }
        [Column(TypeName = "decimal(7, 1)")]
        public decimal addqty { get; set; }
        [Column(TypeName = "decimal(7, 1)")]
        public decimal lessqty { get; set; }
        [Required]
        [StringLength(1)]
        public string endcod { get; set; }
        [Required]
        [StringLength(1)]
        public string cycle_flag { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime update_time { get; set; }
        [StringLength(15)]
        public string updated_by { get; set; }
        // Auto generate in DB, using NotMapped to don't update this
        [Required]
        [StringLength(30)]
        [NotMapped]
        public string Biz_Key { get; set; }
        [Key]
        [StringLength(10)]
        public string wkshno { get; set; }
        [Key]
        [StringLength(10)]
        public string prtno { get; set; }
        [Column(TypeName = "decimal(7, 1)")]
        public decimal wkshqty { get; set; }
        [Column(TypeName = "date")]
        public DateTime mdat { get; set; }
        [StringLength(1)]
        public string uscod { get; set; }
        [StringLength(20)]
        public string remark { get; set; }
    }
}

File: SPA/src/app/_core/models/msQrOrder.ts
export interface MsQrOrder {
    manuf: string;
    brandname: string;
    cusid: string;
    cusna: string;
    kind: string;
    purno: string;
    manno: string;
    eta: string | null;
    bitnbr: string;
    ritnbr: string;
    rmodel: string;
    tolcls: string | null;
    article: string;
    style: string | null;
    size: string;
    tsize: string;
    qty: number;
    pqty: number;
    cqty: number;
    addqty: number;
    lessqty: number;
    endcod: string;
    cycleFlag: string;
    updateTime: string;
    updatedBy: string | null;
    bizKey: string;
    wkshno: string;
    prtno: string;
    wkshqty: number;
    mdat: string;
    uscod: string | null;
}

export interface WkshSumReport {
    mdat_start: string | null;
    mdat_end: string | null;
    close_status: string;
    brandname: string;
    cusna: string;
    manno: string;
    rmodel: string;
    tolcls: string;
    purno: string;
    bitnbr: string;
    kind: string;
    eta_start: string | null;
    eta_end: string | null;
    size: string;
}

export interface SortSumReportParam {
    date_kind: string;
    date_start: string | null;
    date_end: string | null;
    brandname: string;
    cusna: string;
    manno: string;
    purno: string;
    rmodel: string;
    tolcls: string;
    bitnbr: string;
    kind: string;
    eta_start: string | null;
    eta_end: string | null;
    size: string;
}

export interface SortSumReportDTO {
    crday: string | null;
    mdat: string | null;
    brandname: string;
    cusid: string;
    cusna: string;
    manno: string;
    purno: string;
    rmodel: string;
    tolcls: string;
    ritnbr: string;
    bitnbr: string;
    kind: string;
    article: string;
    eta: string | null;
    size: string;
    qty: number | null;
    wkshqty: number | null;
    pqty: number | null;
    sort_qty: number | null;
    diff_qty: number | null;
    cqty: number | null;
}

export interface BrandDTO {
    id: string;
    brandname: string;
}

export interface SortSumDetailReportParam {
    manno: string;
    purno: string;
    size: string;
}

export interface StorageSumDetailReportParam {
    manno: string;
    purno: string;
    size: string;
}
